SH4RE/SH4RE/Messages/MessagesInboxView.swift

```diff
@@ -0,0 +1,200 @@
+                        let name = data["name"] as? String ?? ""
+                        let fromId = data["fromId"] as? String ?? ""
+                        let toId = data["toId"] as? String ?? ""
+                        let timestamp = (data["timestamp"] as? Timestamp)?.dateValue() ?? Date()
+                        let rm = RecentMessage(id: docId, text: text, name:name, fromId: fromId, toId: toId, timestamp: timestamp)
+                        self.recentMessages.insert(rm, at: 0)
+                    }
+                })
+            }
+    }
+    
+    func fetchCurrentUser() {
+        guard let uid = FirebaseManager.shared.auth.currentUser?.uid else {
+            self.errorMessage = "Could not find firebase uid"
+            return
+        }
+        let docRef = Firestore.firestore().collection("User Info").document(uid)
+        docRef.getDocument{ (document, error) in
+            guard error == nil else{
+                print("Error reading document:", error ?? "")
+                return
+            }
+            if let document = document, document.exists {
+                let data = document.data()!
+                let id = document.documentID
+                let name = data["name"] as? String ?? ""
+                FirebaseManager.shared.currentUser = ChatUser(uid:id, name:name)
+            }
+        }
+    }
+    
+    func handleSignOut() {
+        isUserCurrentlyLoggedOut.toggle()
+        try? FirebaseManager.shared.auth.signOut()
+    }
+    
+}
+
+struct MessagesInboxView: View {
+    @Binding var tabSelection: Int
+    @State  var searchQuery: String = ""
+    @EnvironmentObject var currentUser: CurrentUser
+    @ObservedObject var vm = MainMessagesViewModel()
+    @State var shouldNavigateToChatLogView = false
+    var chatLogViewModel = ChatLogViewModel(chatUser: nil)
+    
+    var body: some View {
+
+        ZStack {
+            Color.backgroundGrey.ignoresSafeArea()
+            
+            if (currentUser.isGuest()) {
+                GuestView(tabSelection: $tabSelection).environmentObject(currentUser)
+            }
+            else {
+                NavigationView {
+                    VStack {
+                        customNavBar
+                        messagesView
+                        NavigationLink("", isActive: $shouldNavigateToChatLogView) {
+                            MessagesChat(vm: chatLogViewModel)
+                        }
+                    }
+                }.onAppear(){
+                    vm.fetchCurrentUser()
+                    vm.fetchRecentMessages()
+                }
+            }
+        }
+    }
+    
+    private var customNavBar: some View {
+        HStack(spacing: 16) {
+            VStack(alignment: .leading, spacing: 4) {
+                Text("My Messages")
+                    .font(.title2)
+                    .fontWeight(.bold)
+                
+//                TextField("Search", text: $searchQuery)
```

Comment:
is this just saved for later?

Reply:
yep, i have another ticket to add search